### PR TITLE
chore(gatsby): change HTTP request library from `request` to `got` for future support

### PR DESCRIPTION
**Pull request by ryandrew14**
## Description
This PR changes the HTTP request library used by `gatsby develop` to [`got`](https://www.npmjs.com/package/got#comparison) instead of [`request`](https://www.npmjs.com/package/request). This is based on #13003 which explains that `request` is going into maintenance mode. The PR makes as few changes as possible.

## Related Issues
#13003

### CODE REVIEW

**Comment by ryandrew14** [comment]
Hey Kyle, not sure if I needed to tag you on this (just seeing that "core" is automatically added as a reviewer. Sorry about that!

For whoever reviews this, can I ask why this isn't using a Promise API in the first place and is using streams instead? Is this because Promises are a lot of overhead in Node? Thanks!

File: packages/gatsby/package.json [hunk 1]
```diff
@@ -71,6 +71,7 @@
     "gatsby-react-router-scroll": "^2.0.7",
     "gatsby-telemetry": "^1.0.7",
     "glob": "^7.1.1",
+    "got": "^9.6.0",
```
Comment (pieh) [inline packages/gatsby/package.json]: We can't use `got@^9` because it is Node 8+ and we still support Node 6 (for a little while at least) :(

Comment (ryandrew14) [reply packages/gatsby/package.json]: Great, thanks for the heads-up! I made a quick commit to downgrade.

**Comment by ryandrew14** [comment]
Thanks @sidharthachatterjee! I'm new to Node development -- is the reason we use streams over promises here just that promises are a lot of overhead in Node?

**Comment by sidharthachatterjee** [comment]
@ryandrew14 Thank you for making the contribution!

> is the reason we use streams over promises here just that promises are a lot of overhead in Node?

I wouldn't worry _too_ much about overhead. In this instance, it looks like we're using streams because piping from `req` and to `resp` seems intuitive and elegant!

https://github.com/gatsbyjs/gatsby/blob/master/packages/gatsby/src/commands/develop.js#L213

**Comment by sidharthachatterjee** [comment]
Published in `gatsby@2.3.25`

**Pull request merged**